homeassistant/components/azure_data_explorer/__init__.py

```diff
@@ -0,0 +1,228 @@
+            tuple[int, tuple[datetime, State | None]]
+        ] = asyncio.PriorityQueue()
+        self._listener_remover: Callable[[], None] | None = None
+        self._next_send_remover: Callable[[], None] | None = None
+
+    async def async_start(self) -> None:
+        """Start the component.
+
+        This register the listener and
+        schedules the first send.
+        """
+
+        self._listener_remover = self.hass.bus.async_listen(
+            MATCH_ALL, self.async_listen
+        )
+        self._schedule_next_send()
+
+    async def async_stop(self) -> None:
+        """Shut down the ADX by queueing None, calling send, join queue."""
+        if self._next_send_remover:
+            self._next_send_remover()
+        if self._listener_remover:
+            self._listener_remover()
+        await self._queue.put((3, (utcnow(), None)))
+        await self.async_send(None)
+        await self._queue.join()
+
+    def update_options(self, new_options: dict[str, Any]) -> None:
+        """Update options."""
+        self._send_interval = new_options[CONF_SEND_INTERVAL]
+
+    async def test_connection(self) -> None:
+        """Test the connection to the Azure Data Explorer service."""
+        await self.hass.async_add_executor_job(self._client.test_connection)
+
+    def _schedule_next_send(self) -> None:
+        """Schedule the next send."""
+        if not self._shutdown:
+            self._next_send_remover = async_call_later(
+                self.hass, self._send_interval, self.async_send
+            )
+
+    async def async_listen(self, event: Event) -> None:
+        """Listen for new messages on the bus and queue them for ADX."""
+        if state := event.data.get("new_state"):
+            await self._queue.put((2, (event.time_fired, state)))
+
+    async def async_send(self, _) -> None:
+        """Write preprocessed events to Azure Data Explorer."""
+
+        adx_events = []
+        dropped = 0
+        while not self._queue.empty():
+            _, event = self._queue.get_nowait()
+            adx_event, dropped = self._parse_event(*event, dropped)
+            self._queue.task_done()
+            if adx_event is not None:
+                adx_events.append(adx_event)
+
+        if dropped:
+            _LOGGER.warning(
+                "Dropped %d old events, consider filtering messages", dropped
+            )
+
+        if adx_events:
+            event_string = "".join(adx_events)
+
+            try:
+                await self.hass.async_add_executor_job(
+                    self._client.ingest_data, event_string
+                )
+
+            except KustoServiceError as err:
+                _LOGGER.error("Could not find database or table: %s", err)
+            except KustoAuthenticationError as err:
+                _LOGGER.error("Could not authenticate to Azure Data Explorer: %s", err)
+            except Exception as err:  # pylint: disable=broad-except
```

Comment:
Catching the bare `Exception` outside of config flow is not allowed. Please handle only specific exceptions